src/com/mambu/apisdk/MambuAPIModule.java

```diff
@@ -21,22 +23,26 @@
 	private final String username;
 	private final String password;
 	private final String domain;
+	private final String protocol;
 
 	/***
 	 * Constructor required for setting up the date used for the wrapper to connect to the remote server
 	 * 
+	 * @param protocol
+	 *            the protocol used for communication
 	 * @param domain
 	 *            the domain of the server
 	 * @param username
 	 *            the username required for the connection
 	 * @param password
 	 *            the password required for the connection
 	 */
-	public MambuAPIModule(String domain, String username, String password) {
-
-		this.domain = domain;
-		this.username = username;
-		this.password = password;
+	public MambuAPIModule(Protocol protocol, String domain, String username, String password) {
+		
+		this.domain = Preconditions.checkNotNull(domain, "domain cannot be null");
+		this.username = Preconditions.checkNotNull(username, "username cannot be null");
+		this.password = Preconditions.checkNotNull(password, "password cannot be null");
+		this.protocol = Preconditions.checkNotNull(protocol, "protocol cannot be null").name();

```

Comment:
may want to consider making the "protocol" to be the first line to match params order